[Vertex AI] Add README for unit test mock files

andrewheard opened the pull request:
Added a `README.md` with instructions for running the `update_vertexai_responses.sh` script. This file was also added as a resource in the `FirebaseVertexAIUnit` `testTarget` ensuring that a `Bundle.module` will always be generated by Swift Package Manager. This prevents the following build error if the script has not yet been run:
![Bundle.module Error Screenshot](https://github.com/firebase/firebase-ios-sdk/assets/3010484/2988a89f-df37-4a1f-854b-d17ab9f22d36)

#no-changelog

tanzimfh reviewed (approved):
LGTM, thanks!

Maybe the text in [README.md#firebase-vertex-ai](https://github.com/firebase/firebase-ios-sdk/?tab=readme-ov-file#firebase-vertex-ai) should be replaced with a link to this README?

And should we add the script to [scripts/README.md](https://github.com/firebase/firebase-ios-sdk/blob/main/scripts/README.md) as well?

andrewheard commented:
> Maybe the text in [README.md#firebase-vertex-ai](https://github.com/firebase/firebase-ios-sdk/?tab=readme-ov-file#firebase-vertex-ai) should be replaced with a link to this README?

Agreed. Also I think people are more likely to land on the README in the top-level [FirebaseVertexAI](https://github.com/firebase/firebase-ios-sdk/blob/main/FirebaseVertexAI/README.md) directory so I've moved this new content into there. I updated the README in the `Unit` subdirectory to link to it as well -- easier to maintain if it's all in one place. I should also add a section for the integration tests but will leave that for a follow-up PR.

> And should we add the script to [scripts/README.md](https://github.com/firebase/firebase-ios-sdk/blob/main/scripts/README.md?rgh-link-date=2024-07-04T21%3A23%3A04Z) as well?

Good catch, @tanzimfh! Added one there.
